tests/queries_/test_search.py

```diff
@@ -0,0 +1,837 @@
+        self.assertAlmostEqual(scored.score, 10.0, places=2)
+
+
+@skipUnlessDBFeature("supports_atlas_search")
+class SearchGeoWithinTest(SearchUtilsMixin):
+    def setUp(self):
+        self.create_search_index(
+            Article,
+            "geowithin_location_index",
+            {"mappings": {"dynamic": False, "fields": {"location": {"type": "geo"}}}},
+        )
+        self.article = Article.objects.create(
+            headline="geo", number=2, body="", location={"type": "Point", "coordinates": [40, 5]}
+        )
+        Article.objects.create(
+            headline="geo2", number=3, body="", location={"type": "Point", "coordinates": [-40, -5]}
+        )
+
+    def tearDown(self):
+        self._tear_down(Article)
+        super().tearDown()
+
+    def test_search_geo_within(self):
+        polygon = {
+            "type": "Polygon",
+            "coordinates": [[[30, 0], [50, 0], [50, 10], [30, 10], [30, 0]]],
+        }
+        qs = Article.objects.annotate(
+            score=SearchGeoWithin(
+                path="location",
+                kind="geometry",
+                geo_object=polygon,
+            )
+        )
+        self.wait_for_assertion(lambda: self.assertCountEqual(qs.all(), [self.article]))
+
+    def test_constant_score(self):
+        polygon = {
+            "type": "Polygon",
+            "coordinates": [[[30, 0], [50, 0], [50, 10], [30, 10], [30, 0]]],
+        }
+        constant_score = SearchScoreOption({"constant": {"value": 10}})
+        qs = Article.objects.annotate(
+            score=SearchGeoWithin(
+                path="location",
+                kind="geometry",
+                geo_object=polygon,
+                score=constant_score,
+            )
+        )
+        self.wait_for_assertion(lambda: self.assertCountEqual(qs.all(), [self.article]))
+        scored = qs.first()
+        self.assertAlmostEqual(scored.score, 10.0, places=2)
+
+
+@skipUnlessDBFeature("supports_atlas_search")
+@unittest.expectedFailure
+class SearchMoreLikeThisTest(SearchUtilsMixin):
+    def setUp(self):
+        self.create_search_index(
+            Article,
+            "mlt_index",
+            {
+                "mappings": {
+                    "dynamic": False,
+                    "fields": {"body": {"type": "string"}, "headline": {"type": "string"}},
+                }
+            },
+        )
+        self.article1 = Article.objects.create(
+            headline="Space exploration", number=1, body="Webb telescope"
+        )
+        self.article2 = Article.objects.create(
+            headline="The commodities fall",
+            number=2,
+            body="Commodities dropped sharply due to inflation concerns",
+        )
+        Article.objects.create(
+            headline="irrelevant",
+            number=3,
+            body="This is a completely unrelated article about cooking",
+        )
+
+    def tearDown(self):
+        self._tear_down(Article)
+        super().tearDown()
```

Comment:
You may be able to leverage `TestCase.addCleanup` (or [`addClassCleanup`](https://docs.python.org/3/library/unittest.html#unittest.TestCase.addClassCleanup) per my other comment) to make `create_search_index` register the necessary cleanup.